link/xdp.go

```diff
@@ -0,0 +1,72 @@
+package link
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal/sys"
+)
+
+// XDPAttachTypeFlags represents the XDP attach type.
+type XDPAttachTypeFlags uint32
+
+const (
+	// AttachTypeXDP tries to link as native XDP but in case
+	// the driver does not support native XDP, it will automatically
+	// fall back to generic XDP (default).
+	AttachTypeXDP XDPAttachTypeFlags = 0
```

Comment:
Do we need the `0` flag? I'd expect people to just skip the flag.

Reply:
Zero flag removed.